aten/src/TH/THRandom.c

```diff
@@ -190,9 +191,11 @@ uint64_t THRandom_random(THGenerator *_generator)
 {
   uint64_t y;
 
+  pthread_mutex_lock(&_generator->mutex);
```

Comment:
This will make our CPU PRNGs extremely slow - it will lock and unlock the mutex *for every element in a tensor* (because we usually call `_random` at least once per element).